website/src/pages/[organism]/api/sequences.ts

```diff
@@ -0,0 +1,185 @@
+    const searchParams = getSearchParams(new URL(request.url));
+
+    const organism = cleanOrganism(params.organism);
+    if (organism.organism === undefined) {
+        return new Response(`Organism ${params.organism} not found`, { status: 404 });
+    }
+    const lapisClient = LapisClient.createForOrganism(organism.organism.key);
+
+    const fastaHeaderMapResult = await getAccessionVersionToFastaHeaderMap(lapisClient, searchParams);
+
+    if (fastaHeaderMapResult.isErr()) {
+        return Response.json(fastaHeaderMapResult.error, { status: fastaHeaderMapResult.error.status });
+    }
+
+    const { fastaHeaderMap, dataVersion: detailsDataVersion } = fastaHeaderMapResult.value;
+
+    const sequencesResponse = await lapisClient.streamSequences(searchParams.segment, {
+        ...searchParams.queryFilters,
+        dataFormat: 'ndjson',
+    });
+
+    const sequencesDataVersion = sequencesResponse.headers['lapis-data-version'];
+    if (sequencesDataVersion !== detailsDataVersion) {
+        return Response.json(
+            {
+                type: 'about:blank',
+                title: 'Data version mismatch',
+                status: 409,
+                detail: `Data version mismatch: sequences ${sequencesDataVersion} vs details ${detailsDataVersion}`,
+            } satisfies ProblemDetail,
+            { status: 409 },
+        );
+    }
+
+    return new Response(streamFasta(sequencesResponse, fastaHeaderMap, searchParams), {
+        headers: {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            'Content-Type': 'text/x-fasta',
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            'Content-Disposition': `attachment; filename="${searchParams.downloadFileBasename}.fasta"`,
+        },
+    });
+};
+
+function getSearchParams(url: URL) {
+    const searchParams = url.searchParams;
+
+    const segment = searchParams.get('segment') ?? undefined;
+    const downloadFileBasename = searchParams.get('downloadFileBasename') ?? 'sequences';
+    const headerFields = searchParams.getAll('headerFields');
+    searchParams.delete('segment');
+    searchParams.delete('headerFields');
+    searchParams.delete('downloadFileBasename');
+
+    return {
+        segment,
+        headerFields,
+        downloadFileBasename,
+        queryFilters: Object.fromEntries(searchParams),
+    } satisfies SearchParams;
+}
+
+async function getAccessionVersionToFastaHeaderMap(lapisClient: LapisClient, searchParams: SearchParams) {
+    const details = await lapisClient.getDetails({
+        ...searchParams.queryFilters,
+        fields: [ACCESSION_VERSION_FIELD, ...searchParams.headerFields],
+    });
+
+    return details.map((it) => {
+        const fastaHeaderMap = new Map<string, string>();
+
+        for (const datum of it.data) {
+            const { [ACCESSION_VERSION_FIELD]: accessionVersion, ...rest } = datum;
+            const fastaHeader = Object.values(rest).join('/');
```

Comment:
Let's rather join with `|`, that's also what NCBI Virus uses.

```suggestion
            const fastaHeader = Object.values(rest).join('|');
```